### DISCUSSION TITLE
iOS share sheet and shortcuts integration/REST api help.

### DISCUSSION THREAD

**Discussion by dazeoff** [go-shiori/shiori#349] [Q&A] [answered]
I'm working on an iOS Shortcut to add a share sheet button that add sites to Shiori. I'm currently able to set curl commands that login and pass the url, but I can't figure out how to pass any other info e.g. title, tags, excerpt. If you have an iOS device here is a [link to the shortcut](https://www.icloud.com/shortcuts/6d7e38378f194dd8ae1d00efe6bf6aed) that I currently have as an example. Any guidance on what/if I'm able to use the REST api endpoints to add tags and titles when adding a url?

**Comment by ghost** [comment, marked as answer]
Hi! I saw your issue and tried to do it. It works, kinda. For some reason shiori ignores the title and excerpt when passed with the API, so they do not work (they get fetched automatically by shiori). Tags do work (though they were a nightmare to get working, as Go does not understand JSON lists).

Here is the [iCloud link](https://www.icloud.com/shortcuts/93dce6f60fcd4b1193c44d92f003edc7). When you import it you'll have to fill in your host (`http://<ip>:<port>`), as well as your user and password. It logs in shiori to get a session key, then POSTs the shortcut input to the API. I also added a quick add mode to save time.

Take a look at [this collection](https://github.com/go-shiori/shiori/blob/master/docs/postman/shiori.postman_collection.json) I found in #281, you can import it on Postman for more API examples.

Hope it helps! Cheers!

**Reply by brianreumere** [reply]
This shortcut is great! Thanks for sharing it. I made some small tweaks (get URLs from some additional input types, don't prompt for title or excerpt, accept comma-separated tags, throw an error if a bookmark already exists, show a notification when the bookmark is successfully created). I uploaded it to a repo here: https://github.com/brianreumere/shortcuts/blob/main/Add%20to%20Shiori.shortcut

**Reply by ghost** [reply]
wow, i forgot i made that shortcut! it's been a while hahah. will give yours a try, seems more polished.
maybe for future ios users a link to the shortcut could be added to the docs?

**Reply by fmartingr** [reply]
I have been wondering that, but I'm not sure how that would be in terms of security, since we can preview the action before importing, right? I could enable a repository for the shortcut in the org so you folks could PR into, maybe? I tried making my own shortcut a while ago but failed miserably haha

**Reply by brianreumere** [reply]
You can preview the shortcut (without importing/adding it) by downloading and opening it with the Shortcuts app (on macOS or iOS).

I'd be happy to open a PR into a new repo for it. Unfortunately there's no way to export shortcuts in a text format, so in order to review a PR someone would have to open the shortcut and preview it.

**Reply by pythoninthegrass** [reply]
@brianreumere Thank you so much for setting this up!

It will work brilliantly as a stopgap while I dabble in Flet to create a dedicated iOS app for Shiori.

Also lays the groundwork for using the API to save new pages <3

**Reply by fmartingr** [reply]
>  I dabble in Flet to create a dedicated iOS app for Shiori.

Are you working on a iOS shiori app 👀

**Reply by pythoninthegrass** [reply]
That's the plan @fmartingr 

It'll be pretty utilitarian as I'm not a mobile dev, but I really like Shiori and have been let down by Pocket and its derivatives lately.

Once I get an mvp up and running, I'll invite you as a collaborator ☺️

(I have a lot of incentive and a bit of bandwidth, so it'll be available hopefully by the end of July.)

**Reply by fmartingr** [reply]
Please ping me when you have something testable. Also take a look at #466. Hopefully I wont break many things!

**Comment by ghost** [comment]
I wrote [this wiki page](https://github.com/go-shiori/shiori/blob/master/docs/API.md) explaining the API a bit

Edit: Updated link to new wiki location

**Comment by Fletcher-Alderton** [comment]
Sorry to open this up again but I made an improved version which automatically inputs the title and description using regex to extract it from the website. [link to shortcut](https://www.icloud.com/shortcuts/5e1479fde4e4404fbf6d5adb92e17343)
I’m having an issue where some sites have large spaces at the start and end of the title and i believe that’s what’s causing issue, if i could have some conformation on if this is the case with the api that would be great.

**Reply by fmartingr** [reply]
@Fletcher-Alderton I have removed your link since it contained your Shiori instance url and credentials. Please update your password and share a duplicate of the shortcut without sensitive data. Also if you can, unshare the link as well since it will be present in email notifications.

**Reply by Fletcher-Alderton** [reply]
Thanks for that, completely crossed my mind.

**Comment by yusing** [comment]
I made another [Shortcut](https://www.icloud.com/shortcuts/ba5566bacde147d0ba9210be193e09ce) which uses the webpage title directly but no excerpts

**Comment by dertuxmalwieder** [comment]
I presume that all of these will be deprecated with the arrival of APIv1?
